### ISSUE TITLE
Typed HubContext Clients property

### ISSUE THREAD

**Issue by kant2002** [aspnet/SignalR#722]
I have application which want to translate existing SignalR 2.2 hubs using new .NET Core version. I order to retranslate notifications, I have to use `HubContext` if I understand current version properly. But current version of `HubContext<T>` provide untyped version of `IHubClients` which is require me to manually write all notifications. This is easy way to introduce stupid errors. I would like to have some way to have access to `IHubClients<T>` outside of `Hub` implementations. Do you have any suggestions how I could do that?

**Comment by FTWinston** [comment]
Yeah I would also find this useful, I imagine many others would too. I'm far from an expert, but I've made a pull request that implements this, and it seems to work for me:
https://github.com/aspnet/SignalR/pull/783

**Comment by billbogaiv** [comment]
I'd like to see #783 get implemented. My current approach uses `nameof` just to avoid direct string-usage:

```csharp
public Task DoThing()
{
    Clients.All.InvokeAsync(nameof(IThing.DoIt));
}
```

**Comment by moozzyk** [comment]
If you derive your hub from `Hub<T>` you the `Clients` property will be of `IHubClients<T>` and you will be able to use methods defined on `T` instead of `strings`.

**Comment by davidfowl** [comment]
It's about using it outside of the hub itself. It's a gap we currently have to fill.

**Comment by FTWinston** [comment]
See now #890

**Comment by moozzyk** [comment]
Fixed with a7fb24350192f76e3539e60231d89a90eef7c13e
